libp2p-swarm/src/transport.rs

```diff
@@ -707,6 +708,13 @@ where
 		}
 	}
 
+	/// Turns this upgraded node into a `ConnectionReuse`. If the `Output` implements the
+	/// `StreamMuxer` trait, the returned object will implement `Transport` and `MuxedTransport`.
+	#[inline]
+	pub fn into_connection_reuse(self) -> ConnectionReuse<T, C> {
```

Comment:
I wish `Into` was a trait without type parameters so `.into::<ConnectionReuse<_, _>>()` was valid syntax, but it's a major breaking change now.